### DISCUSSION TITLE
Slow loading of data

### DISCUSSION THREAD

**Discussion by stellarfinn** [Budibase/budibase#9210] [Help]
Hello, 

I need help figuring out why my data loads so slowly. I'm assuming its in how I'm using the data providers. My budibase is ciaoimports.budibase.app. 

I have three databases and need to access one, two or all three on a single page in order to display certain information. I also have javascript calculations to manipulate some data in repeaters. 

Is there anyone with Budibase that can look at my app and tell me what I'm doing that's bogging it down so much?

**Comment by melohagan** [comment]
Hey @stellarfinn 

There's currently an issue in the latest release with duplicate queries that's currently being worked on: https://github.com/Budibase/budibase/issues/9143

**Reply by stellarfinn** [reply]
My issue existed prior to the latest release. It's worse now for sure.
